listenbrainz/webserver/static/js/src/user/Listens.tsx

```diff
@@ -421,23 +430,56 @@ export default class Listens extends React.Component<
     if (!feedback) {
       return;
     }
-    const recordingFeedbackMap: RecordingFeedbackMap = {};
+    const recordingMsidFeedbackMap: RecordingFeedbackMap = {};
+    const recordingMbidFeedbackMap: RecordingFeedbackMap = {};
     feedback.forEach((fb: FeedbackResponse) => {
-      recordingFeedbackMap[fb.recording_msid] = fb.score;
+      if (fb.recording_msid) {
+        recordingMsidFeedbackMap[fb.recording_msid] = fb.score;
+      }
+      if (fb.recording_mbid) {
+        recordingMbidFeedbackMap[fb.recording_mbid] = fb.score;
+      }
     });
-    this.setState({ recordingFeedbackMap });
+    this.setState({ recordingMsidFeedbackMap, recordingMbidFeedbackMap });
   };
 
   updateFeedback = (
     recordingMsid: string,
-    score: ListenFeedBack | RecommendationFeedBack
+    score: ListenFeedBack | RecommendationFeedBack,
+    recordingMbid?: string
   ) => {
-    const { recordingFeedbackMap } = this.state;
-    const newFeedbackMap = {
-      ...recordingFeedbackMap,
-      [recordingMsid]: score as ListenFeedBack,
-    };
-    this.setState({ recordingFeedbackMap: newFeedbackMap });
+    const { recordingMsidFeedbackMap, recordingMbidFeedbackMap } = this.state;
+
+    if (recordingMsid && recordingMbid) {
+      const newMsidFeedbackMap = {
+        ...recordingMsidFeedbackMap,
+        [recordingMsid]: score as ListenFeedBack,
+      };
+      const newMbidFeedbackMap = {
+        ...recordingMbidFeedbackMap,
+        [recordingMbid]: score as ListenFeedBack,
+      };
+      this.setState({
+        recordingMsidFeedbackMap: newMsidFeedbackMap,
+        recordingMbidFeedbackMap: newMbidFeedbackMap,
+      });
+    }
+
+    if (recordingMsid) {
+      const newMsidFeedbackMap = {
+        ...recordingMsidFeedbackMap,
+        [recordingMsid]: score as ListenFeedBack,
+      };
+      this.setState({ recordingMsidFeedbackMap: newMsidFeedbackMap });
+    }
+
+    if (recordingMbid) {
+      const newMbidFeedbackMap = {
+        ...recordingMbidFeedbackMap,
+        [recordingMbid]: score as ListenFeedBack,
+      };
+      this.setState({ recordingMbidFeedbackMap: newMbidFeedbackMap });
+    }
```

Comment:
Any specific reason to do this in two separate way (mbid+msid together, and msid or mbid separate) rather than like this all in one go?

```suggestion
   	let newMsidFeedbackMap = {...recordingMsidFeedbackMap};
	let newMbidFeedbackMap = {...recordingMbidFeedbackMap};
	
    if (recordingMsid) {
		newMsidFeedbackMap[recordingMsid]= score as ListenFeedBack;
    }
    if (recordingMbid) {
		recordingMbidFeedbackMap[recordingMbid]= score as ListenFeedBack;
    }
	this.setState({ recordingMsidFeedbackMap: newMsidFeedbackMap, recordingMbidFeedbackMap: newMbidFeedbackMap });
```

Reply:
I was a bit confused on it initially, whether it would cause unnecessary re-renders but thinking more now, doing it in 1 go should be fine. Thanks! :D

Reply:
I figured that was probably the case, and realized we could do it all with a single setState call (and IIRC trying to set an exact clone of an object in the state shouldn't trigger a re-render [?])
In any case, we'll need to re-render because one of these dicts is going to be updated, so probably inevitable.

Soon enough I want to redo the whole feedback dict thing anyway, to avoid having it implemented separately in each page and using a React context instead to share the state across the app (and save some API calls too eventually)

Reply:
Done. Makes sense